packages/parser/src/analyze-scope.ts

```diff
@@ -344,6 +344,17 @@ class Referencer extends TSESLintScope.Referencer<ScopeManager> {
     node.arguments.forEach(this.visit, this);
   }
 
+  /**
+   * Visit optional member expression.
+   * @param node The OptionalMemberExpression node to visit.
+   */
+  OptionalMemberExpression(node: TSESTree.OptionalMemberExpression): void {
```

Comment:
should also add support for `OptionalCallExpression` in this PR
(i.e. pull in your change from #1175)

And add tests accordingly

Reply:
Done